### DISCUSSION TITLE
[Feature request] interactive/expandable code comments-explanations

### DISCUSSION THREAD

**Discussion by camilogarciabotero** [quarto-dev/quarto-cli#3490] [Feature Requests]
Hi,

I was watching the [docs](https://nightly.mealie.io/documentation/getting-started/installation/postgres/) of an app and came across a nice feature for commenting code lines with an interactive, expandable icon: here

https://user-images.githubusercontent.com/25586362/204315187-2a5ad145-3a93-47a2-a7df-93988d589532.mov

Would be nice to have such a feature in Quarto!

**Comment by mcanouil** [comment]
This looks like Nutshell.js which has an Quarto extension: <https://github.com/schochastics/quarto-nutshell>

@schochastics did you try to use it inside a code cell?

**Reply by camilogarciabotero** [reply]
Yeah, `quarto-nutshell` also came to my mind! Not sure if the native Nutshell.js can handle code comments either.

**Comment by schochastics** [comment]
this looks pretty cool, but I do not think my extension can handle this. I at least failed just now

**Comment by camilogarciabotero** [comment]
This was recently implemented here: https://github.com/quarto-dev/quarto-cli/pull/3676 it will be awesome to create better resources! Thanks @dragonstyle I tried the feature and is super: https://twitter.com/Gaspardelanoche/status/1611044978452463619?s=20&t=IxK5oGTnQ5dDDE-7EwyUYw

**Comment by mcanouil** [comment]
Closing as completed, see <https://quarto.org/docs/authoring/code-annotation.html>.
